src/StackExchange.Redis/Configuration/DefaultOptionsProvider.cs

```diff
@@ -0,0 +1,197 @@
+        public virtual bool CheckCertificateRevocation => true;
+
+        /// <summary>
+        /// The number of times to repeat the initial connect cycle if no servers respond promptly.
+        /// </summary>
+        public virtual int ConnectRetry => 3;
+
+        /// <summary>
+        /// Specifies the time that should be allowed for connection.
+        /// Falls back to Max(5000, SyncTimeout) if null.
+        /// </summary>
+        public virtual TimeSpan? ConnectTimeout => null;
+
+        /// <summary>
+        /// The command-map associated with this configuration.
+        /// </summary>
+        public virtual CommandMap CommandMap => null;
+
+        /// <summary>
+        /// Channel to use for broadcasting and listening for configuration change notification.
+        /// </summary>
+        public virtual string ConfigurationChannel => "__Booksleeve_MasterChanged";
+
+        /// <summary>
+        /// The server version to assume.
+        /// </summary>
+        public virtual Version DefaultVersion => RedisFeatures.v3_0_0;
+
+        /// <summary>
+        /// Specifies the time interval at which connections should be pinged to ensure validity.
+        /// </summary>
+        public virtual TimeSpan KeepAliveInterval => TimeSpan.FromSeconds(60);
+
+        /// <summary>
+        /// Type of proxy to use (if any); for example <see cref="Proxy.Twemproxy"/>.
+        /// </summary>
+        public virtual Proxy Proxy => Proxy.None;
+
+        /// <summary>
+        /// The retry policy to be used for connection reconnects.
+        /// </summary>
+        public virtual IReconnectRetryPolicy ReconnectRetryPolicy { get; internal set; }
+
+        /// <summary>
+        /// Indicates whether endpoints should be resolved via DNS before connecting.
+        /// If enabled the ConnectionMultiplexer will not re-resolve DNS when attempting to re-connect after a connection failure.
+        /// </summary>
+        public virtual bool ResolveDns => false;
+
+        /// <summary>
+        /// Specifies the time that the system should allow for synchronous operations.
+        /// </summary>
+        public virtual TimeSpan SyncTimeout => TimeSpan.FromSeconds(5);
+
+        /// <summary>
+        /// Tie-breaker used to choose between masters (must match the endpoint exactly).
+        /// </summary>
+        public virtual string TieBreaker => "__Booksleeve_TieBreak";
+
+        /// <summary>
+        /// Check configuration every n interval.
+        /// </summary>
+        public virtual TimeSpan ConfigCheckInterval => TimeSpan.FromMinutes(1);
+
+        // Note: this is statically backed because it doesn't change.
+        private static string defaultClientName;
+        /// <summary>
+        /// The default client name for a connection, with the library version appended.
+        /// </summary>
+        public virtual string ClientName =>
+            defaultClientName ??= (TryGetAzureRoleInstanceIdNoThrow()
```

Comment:
For example if you're running a _client_ in Azure but running your own Redis or at another host, this would regress from role ID down to default client name. I'm not sure how many users are in that boat but was trying to avoid that case regression - thoughts?

Reply:
Ah yes, I knew I was forgetting a scenario :)